### PR TITLE
#164069295 Chore -unit test for all functionality

### PR DESCRIPTION
**Pull request by akram256**
**What does this PR do?**
* This tests functionality of all functionality in the codebase

**Description of the tasks to be completed?**
* Tests for the existing functionality which is log in and registration of a user have been added to the code base.
* Tests for missing username, missing email, login with non-existing user

**How should this be manually tested?**

* Run  `python manage.py test`

**What are the relevant pivot tracker stores?**
#164069297

 **Screenshots**
* Below are screenshots of the running test

screenshot 2019-02-25 at 19 32 37

screenshot 2019-02-25 at 19 35 32

### CODE REVIEW

**Comment by walimike** [comment]
LGTM

File: authors/apps/authentication/tests/test_functionality.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertIn("This field may not be blank.", str(response.data))
+
+    def test_login(self):
+        # url = reverse('signup')
+        data = {"user": {"username": "akram",
+                         "email": "akram@gmail.com", "password":
+                         "akrammukasa"}}
+        response = self.client.post('/api/users/', data, format='json')
+        # url = reverse('login')
```
Comment (Stanley-Okwii) [inline authors/apps/authentication/tests/test_functionality.py]: Why is this commented out, if it's not needed then remove it? @akram256

File: authors/apps/authentication/tests/test_functionality.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
+from django.urls import reverse
+from rest_framework.test import APITestCase
+from rest_framework import status
+from authors.apps.authentication.models import User
+
+
+class Authentication(APITestCase):
+
+    def test_for_new_user(self):
+        # url = reverse('signup')
```
Comment (Stanley-Okwii) [inline authors/apps/authentication/tests/test_functionality.py]: Be consistent with your comments on functions, don't leave other functions without comments

File: authors/apps/authentication/tests/test_functionality.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                         "email": "akram@gmail.com", "password":
+                         "akrammukasa"}}
+        response = self.client.post('/api/users/', data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_for_missing_username(self):
+        """
+        Method for testing if there is a missing username during registration.
+        """
+        data = {"user": {"username": "", "email": "akram@gmail.com",
```
Comment (Stanley-Okwii) [inline authors/apps/authentication/tests/test_functionality.py]: Be consistent with either double quotes or single quotes

**Review by Stanley-Okwii — COMMENTED** [review]
Good work @akram256

**Comment by PatrickCmd** [comment]
@akram256 Can we have only one commit for the PR instead of two.

**Comment by PatrickCmd** [comment]
@akram256 Move the tests package to the root of the project and rename the test file to **test_authentication**.

**Pull request merged**